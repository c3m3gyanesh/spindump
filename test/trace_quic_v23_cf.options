--report-notes --report-spins --format json
