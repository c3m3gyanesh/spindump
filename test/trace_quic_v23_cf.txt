A test between the Cloudflare QUIC implementation (server) and the Picoquic implementation (client). And first attempting the wrong port without response. Using the following commands

    ./picoquic/picoquicdemo -l ./test.log -L -a "h3-23" -v ff000017 cloudflare-quic.com 4433 /50000
    ./picoquic/picoquicdemo -l ./test.log -L -a "h3-23" -v ff000017 cloudflare-quic.com 443 /50000
