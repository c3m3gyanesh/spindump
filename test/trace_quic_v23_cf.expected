[
{ "Event": "new", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "08092f5f08d37172-02ef666c1bda2e05 (65442:4433)", "Ts": 1571177826777310, "State": "Starting", "Notes": "No rsp,v23,no spin", "Packets1": 0, "Packets2": 0, "Bytes1": 0, "Bytes2": 0 },
{ "Event": "new", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "08f5043ebb92c47b-63756457211ff7bb (57085:443)", "Ts": 1571177835699975, "State": "Starting", "Notes": "No rsp,v23,no spin", "Packets1": 0, "Packets2": 0, "Bytes1": 0, "Bytes2": 0 },
{ "Event": "measurement", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "08f5043ebb92c47b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835710747, "State": "Up", "Notes": "No rsp,v23,no spin", "Right_rtt": 10772, "Packets1": 1, "Packets2": 0, "Bytes1": 1280, "Bytes2": 0, "Bandwidth1": 1280, "Bandwidth2": 0 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835746882, "State": "Up", "Notes": "V23,no spin", "Value": 0, "Who": "initiator", "Packets1": 6, "Packets2": 8, "Bytes1": 2962, "Bytes2": 5967, "Bandwidth1": 2962, "Bandwidth2": 5967 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835746984, "State": "Up", "Notes": "V23,no spin", "Value": 0, "Who": "initiator", "Packets1": 7, "Packets2": 8, "Bytes1": 3087, "Bytes2": 5967, "Bandwidth1": 3087, "Bandwidth2": 5967 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835755601, "State": "Up", "Notes": "V23,no spin", "Value": 0, "Who": "responder", "Packets1": 8, "Packets2": 9, "Bytes1": 4587, "Bytes2": 6053, "Bandwidth1": 4587, "Bandwidth2": 6053 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835755606, "State": "Up", "Notes": "V23,no spin", "Value": 0, "Who": "responder", "Packets1": 8, "Packets2": 10, "Bytes1": 4587, "Bytes2": 6567, "Bandwidth1": 4587, "Bandwidth2": 6567 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835755607, "State": "Up", "Notes": "V23,no spin", "Value": 0, "Who": "responder", "Packets1": 8, "Packets2": 11, "Bytes1": 4587, "Bytes2": 6626, "Bandwidth1": 4587, "Bandwidth2": 6626 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835755608, "State": "Up", "Notes": "V23,no spin", "Value": 0, "Who": "responder", "Packets1": 8, "Packets2": 12, "Bytes1": 4587, "Bytes2": 6685, "Bandwidth1": 4587, "Bandwidth2": 6685 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835755914, "State": "Up", "Notes": "V23,no spin", "Value": 1, "Who": "initiator", "Packets1": 8, "Packets2": 13, "Bytes1": 4587, "Bytes2": 6769, "Bandwidth1": 4587, "Bandwidth2": 6769 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835756669, "State": "Up", "Notes": "V23,no R-spin", "Value": 0, "Who": "responder", "Packets1": 9, "Packets2": 13, "Bytes1": 4670, "Bytes2": 6769, "Bandwidth1": 4670, "Bandwidth2": 6769 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835846163, "State": "Up", "Notes": "V23,no R-spin", "Value": 1, "Who": "initiator", "Packets1": 9, "Packets2": 14, "Bytes1": 4670, "Bytes2": 6829, "Bandwidth1": 4670, "Bandwidth2": 6829 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835945584, "State": "Up", "Notes": "V23,no R-spin", "Value": 1, "Who": "initiator", "Packets1": 10, "Packets2": 14, "Bytes1": 6170, "Bytes2": 6829, "Bandwidth1": 6170, "Bandwidth2": 6829 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177835956176, "State": "Up", "Notes": "V23,no R-spin", "Value": 0, "Who": "responder", "Packets1": 11, "Packets2": 14, "Bytes1": 7598, "Bytes2": 6829, "Bandwidth1": 7598, "Bandwidth2": 6829 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177836425755, "State": "Up", "Notes": "V23,no R-spin", "Value": 0, "Who": "responder", "Packets1": 11, "Packets2": 15, "Bytes1": 7598, "Bytes2": 6892, "Bandwidth1": 7598, "Bandwidth2": 6892 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177836426015, "State": "Up", "Notes": "V23,no R-spin", "Value": 1, "Who": "initiator", "Packets1": 11, "Packets2": 16, "Bytes1": 7598, "Bytes2": 7440, "Bandwidth1": 7598, "Bandwidth2": 7440 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177836426348, "State": "Up", "Notes": "V23,no R-spin", "Value": 0, "Who": "responder", "Packets1": 12, "Packets2": 16, "Bytes1": 7675, "Bytes2": 7440, "Bandwidth1": 7675, "Bandwidth2": 7440 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177836426424, "State": "Up", "Notes": "V23,no R-spin", "Value": 1, "Who": "initiator", "Packets1": 12, "Packets2": 17, "Bytes1": 7675, "Bytes2": 7498, "Bandwidth1": 7675, "Bandwidth2": 7498 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177836427808, "State": "Up", "Notes": "V23,no R-spin", "Value": 0, "Who": "responder", "Packets1": 13, "Packets2": 17, "Bytes1": 7747, "Bytes2": 7498, "Bandwidth1": 7747, "Bandwidth2": 7498 },
{ "Event": "spinvalue", "Type": "QUIC", "Addrs": ["172.16.114.251","104.20.107.50"], "Session": "f5043ebb92c47b2b-14da2d95dc13636e2187f73ba4c44f4391aaca3d (57085:443)", "Ts": 1571177836427890, "State": "Up", "Notes": "V23,no R-spin", "Value": 1, "Who": "initiator", "Packets1": 13, "Packets2": 18, "Bytes1": 7747, "Bytes2": 7556, "Bandwidth1": 7747, "Bandwidth2": 7556 }
]
